src/main/java/com/jvm_bloggers/core/rss/converters/SyndFeedToJsonConverter.java

```diff
@@ -71,12 +73,18 @@ private static JSONObject toJson(SyndEntry entry) {
         return json;
     }
 
+    private static void setLinkIfPresent(SyndFeed feed, JSONObject json) {
+        Stream.ofAll(feed.getLinks())
+            .find(link -> SELF_REL.equals(link.getRel()))
+            .map(SyndLink::getHref)
+            .forEach(link -> json.put(LINK, link));
+    }
+
     public JSONObject convert(SyndFeed feed) {
         log.debug("Building JSON from the RSS feed...");
 
         JSONObject json = toJson(feed);
         json.put(GENERATOR, baseUrl);
-        json.put(LINK, baseUrl + BlogPostsController.RSS_FEED_MAPPING);
```

Comment:
As stated previously, `link` was never set. 
Here it was overwritten with `jvm-bloggers.com/pl/rss` instead of `jvm-bloggers.com/pl/rss.json` . I've changed this to extract `self` link that should always be present.